src/components/RenderPipeline/RenderPipelineV2.tsx

```diff
@@ -0,0 +1,74 @@
+import { useStore } from "@/src/types/StoreContext";
+import { MergeNode } from "@/src/components/RenderPipeline/MergeNode";
+import { useRenderTarget, useRenderTargets } from "@/src/hooks/renderTarget";
+import { observer } from "mobx-react-lite";
+import { WebGLRenderTarget } from "three";
+import { BlockStackNode } from "./BlockStackNode";
+import { LayerV2 } from "@/src/types/Layer/LayerV2";
+
+type Props = {
+  renderTargetZ: WebGLRenderTarget;
+};
+
+export const RenderPipelineV2 = observer(function RenderPipeline({
+  renderTargetZ,
+}: Props) {
+  const store = useStore();
+  const layers = store.layers as LayerV2[];
+  if (layers.length === 0) return null;
+  return <LayerNode layer={layers[0]} destinationTarget={renderTargetZ} />;
+});
+
+type LayerNodeProps = {
+  layer: LayerV2;
+  destinationTarget: WebGLRenderTarget;
+};
+
+function LayerNode({ layer, destinationTarget }: LayerNodeProps) {
+  const renderTargets = useRenderTargets(layer);
```

Comment:
I'm seeing that we create our render targets here inside of the layer node as well as inside of the merge node. We may want to move these creations to the top level of the pipeline and pass down render targets as props to minimize deletion/recreation for instance when the number of active layers goes from 2 to 1 and back to 2.